### PR TITLE
Upgrade to Elm 0.18

### PR DESCRIPTION
**Pull request by benbrandt**
This package has some great functions, so I wanted to be able to use it in a project using 0.18.

I was using elm-format, so a lot of the changes are from that, let me know if you'd rather I not use it on this project.

I was still getting errors on `performFailproof` though, so if you don't mind giving me any pointers on how to get that function to work, I'd appreciate it! Thanks!

### CODE REVIEW

File: src/Task/Extra.elm [hunk 1]
```diff
@@ -80,7 +92,11 @@ performFailproof : (a -> msg) -> Task Never a -> Cmd msg
 performFailproof =
```
Comment (jtojnar) [inline src/Task/Extra.elm]: This, along with `never`, can be removed since it was added to core.

Comment (benbrandt) [reply src/Task/Extra.elm]: Ok awesome thank you!

**Comment by benbrandt** [comment]
This has been updated with `performFailproof` and `never` removed thanks to @jtojnar's advice.

Let me know if you want anything else changed, thanks!

**Comment by avh4** [comment]
Thanks, we are currently in the process of trying to move this package to elm-community.  See https://github.com/elm-community/Manifesto/issues/62

**Comment by benbrandt** [comment]
Great! Let me know if you need anything else from me

**Comment by avh4** [comment]
There was some pushback on elm-community....   So I'm curious now, what functions from this package to you actually need?   There doesn't seem to be much useful left after `never` and `performFailproof` were removed.

**Comment by benbrandt** [comment]
The parts I actually am using are `optional` and `delay`

My use case may not be widespread enough to warrant merging into an elm-community package though. Core didn't have this functionality (as far as I could tell)

**Comment by avh4** [comment]
Got it.  Yeah, I think I'm going to suggest you just move `optional` and `delay` into your own project, and we'll deprecate this project.

**Comment by benbrandt** [comment]
Ok thanks for the help on this!

**Pull request closed without merging**